### DISCUSSION TITLE
need help with Hamburger menu in react

### DISCUSSION THREAD

**Discussion by chris-yooo** [neuefische/questions#41] [Web] [answered]
I try to use Hamburger from hamburger-react but i cant display my Navlinks into the Menu - the Menu self works but didnt display after onclick ...

![Screenshot (2)](https://user-images.githubusercontent.com/55069641/182372242-5f93fb25-5934-461e-85b9-18eb742f4140.png)

can anyone help me in room 1 ?

**Comment by pixelass** [comment]
Du kannst in der Toggle Funktion kein return haben.

Stattdessen solltest du einen State anlegen.

Das konnte ungefähr so aussehen:  (`{isOpen ? "Menu is open" : "Menu is closed"}` ist hier nur exemplarisch)

```jsx
function MyComponent() {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <div>
      <button onClick={() => {
        setIsOpen(!isOpen);
      }}>Toggle Menu</button>
      {isOpen ? "Menu is open" : "Menu is closed"} 
    </div>
  )
}

**Comment by chris-yooo** [comment, marked as answer]
diese Hamburger Menu animation hab ich von hier genommen: https://hamburger-react.netlify.app/

dort steht das beispielt das ich im code hatte mit drinnen :D

ich hab es so gelöst mit stefan ->
![hh](https://user-images.githubusercontent.com/55069641/182556922-5cb50ea1-969e-480f-bc2e-d5f6e891f87e.png)
![41](https://user-images.githubusercontent.com/55069641/182556934-85d94b1e-390c-478c-8bdc-8a9cd03275a1.png)
![79](https://user-images.githubusercontent.com/55069641/182556941-690de9a4-cd10-4442-a0b2-7f16cb992bf0.png)

**Reply by pixelass** [reply]
Sehr schön. Ist aber am Ende genau das was ich gemaint habe. In deinem ersten Beispiel hast du innerhalb der `onToggle` Funktion ein `return` gehabt. Das war da der Fehler:

<img width="516" alt="Screenshot_03_08_22__10_45" src="https://user-images.githubusercontent.com/1148334/182565757-8095907c-28f1-45e7-978d-2ed68d803cd5.png">
